cou/apps/base.py

```diff
@@ -857,7 +857,10 @@ def _check_mismatched_versions(self, units: Optional[list[Unit]]) -> None:
         :raises MismatchedOpenStackVersions: When the units of the app are running
                                              different OpenStack versions.
         """
-        if units:
+        # NOTE (gabrielcocenza) nova-compute is upgraded using paused-single-unit,
+        # so it's possible to have mismatched version in applications units that are
+        # nova-compute or colocated with it.
+        if any("nova-compute" in machine.apps for machine in self.machines.values()):
```

Comment:
is `machine.apps` a list of charm names (instead of a list of application classes or a list of application names)?

Reply:
I thought it was charm name, but It's actually a tuple of application names that are in a specific machine. See:
https://github.com/canonical/charmed-openstack-upgrader/blob/1d25afc8306595eb549438ed39ce4269df6797d4/cou/utils/juju_utils.py#L283-L296

I'll need to change the logic a little bit. Thanks

Reply:
I do not think that this is good approach, we want to avoid any libjuju object passing to our code, so Machine should not include juju.application.Application object.

Reply:
Och wait the `unit.application` in only name here not an Application object?

Reply:
It's just the name of the application and not a juju object. I guess I'll need to include the change as we almost done [before](https://github.com/canonical/charmed-openstack-upgrader/pull/248/commits/87189c184e0fa4185a89971e671d9adf263db335#r1497891237)

Reply:
You can get applications and find the charm name there.